elasticsearch/src/http/transport.rs

```diff
@@ -595,16 +629,122 @@ impl CloudConnectionPool {
 
 impl ConnectionPool for CloudConnectionPool {
     /// Gets a reference to the next [Connection]
-    fn next(&self) -> &Connection {
-        &self.connection
+    fn next(&self) -> Connection {
+        self.connection.clone()
+    }
+}
+
+/// A Connection Pool that manages a static connection of nodes
+#[derive(Debug, Clone)]
+pub struct MultiNodeConnectionPool<TStrategy = RoundRobin> {
```

Comment:
If we rename the `Strategy` trait to `ConnectionSelector` or something similar we can avoid the `T` prefix and name this `Strategy` (or `Selector` with the suggested name).